server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteServicesImpl.java

```diff
@@ -38,23 +38,23 @@ public CheckinNote save(@NotNull CheckinNote checkinNote) {
         boolean isAdmin = currentUserServices.isAdmin();
         boolean isPdl = currentUserServices.hasRole(RoleType.PDL);
 
-        if (!isAdmin && !isPdl) {
-            throw new PermissionException("You do not have permission to access this resource");
-        }
-
         final UUID checkinId = checkinNote.getCheckinid();
         final UUID createById = checkinNote.getCreatedbyid();
         CheckIn checkinRecord = checkinRepo.findById(checkinId).orElse(null);
         Boolean isCompleted = checkinRecord != null ? checkinRecord.isCompleted() : null;
-        final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+
+        if (checkinRecord!=null && !isAdmin && !currentUser.getId().equals(checkinRecord.getTeamMemberId()) && !currentUser.getId().equals(checkinRecord.getPdlId())) {
```

Comment:
This is missing a check. Access should also be allowed if the currentUser is the pdl for the team member whose checkin this is.

Reply:
@Maroju100 was this check added ?